voxblox/src/integrator/esdf_integrator.cc

```diff
@@ -65,9 +65,11 @@ void EsdfIntegrator::addNewRobotPosition(const Point& position) {
         continue;
       }
       EsdfVoxel& esdf_voxel = block_ptr->getVoxelByVoxelIndex(voxel_index);
-      if (!esdf_voxel.observed) {
+      // We can clear unobserved or hallucinated voxels.
+      if (!esdf_voxel.observed || esdf_voxel.hallucinated) {
         esdf_voxel.distance = config_.default_distance_m;
         esdf_voxel.observed = true;
+        esdf_voxel.hallucinated = true;
```

Comment:
maybe consider adding this new behavior to the description/comment of this function?